controller/golden_files_test.go

```diff
@@ -59,8 +59,14 @@ func testableCompareWithGolden(update bool, goldenFile string, actualObj interfa
 		return errs.WithStack(err)
 	}
 	if update {
+		// Make sure the directory exists where to write the file to
+		dir := filepath.Dir(absPath)
+		err := os.MkdirAll(dir, os.FileMode(0777))
```

Comment:
don't you need (or want) to check if the `dir` exists and is a directory with the needed write access before calling `os.MkdirAll` ?

Reply:
I did check that but the documentation for `os.MkdirAll` says in the last sentence that it does that for me:

```go
// MkdirAll creates a directory named path,
// along with any necessary parents, and returns nil,
// or else returns an error.
// The permission bits perm are used for all
// directories that MkdirAll creates.
// If path is already a directory, MkdirAll does nothing
// and returns nil.
func MkdirAll(path string, perm FileMode) error {
```

And if I don't have write access the write operation function will tell me so. So it essentially is the same as checking for write access and telling: "no can't do".

Reply:
ok, fair enough ;)